app/controllers/pieces_controller.rb

```diff
@@ -19,7 +19,7 @@ def update
   private
   
     def set_piece
-      @piece = Piece.find(params[:id])
+      @piece ||= Piece.find(params[:id])
```

Comment:
Ah, OK. I didn't know these things. I'm not aware that it's called multiple times, so you're obviously correct that the ||= is unnecessary. Thanks for pointing this out.